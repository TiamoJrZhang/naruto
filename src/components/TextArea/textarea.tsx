import React, {Component} from 'react'

type KeyboardHandler = React.KeyboardEventHandler<HTMLTextAreaElement>

export interface TextAreaProps {
  onPressEnter?: KeyboardHandler
  onKeyDown?: KeyboardHandler
  maxLength?: number
  disabled?: boolean
}

class Textarea extends Component<TextAreaProps> {
  constructor(props) {
    super(props)
    this.justTest = this.justTest.bind(this)
  }

  justTest = () => {
    console.log('this...', this)
  }

  handleKeyDown = (e: React.KeyboardEvent<HTMLTextAreaElement>) => {
    this.justTest()
    const {onPressEnter, onKeyDown} = this.props
    if (e.keyCode === 13 && onPressEnter) {
      onPressEnter(e)
    }
    if (onKeyDown) {
      onKeyDown(e)
    }
  }

  render() {
    const {maxLength, disabled} = this.props
    return (
      <div>
        <textarea onKeyDown={this.handleKeyDown} maxLength={maxLength} disabled={disabled} />
      </div>
    )
  }
}

export default Textarea
