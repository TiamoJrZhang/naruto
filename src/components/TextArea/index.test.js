/** @format */

import React from 'react'
import {mount} from 'enzyme'
import TextArea from './textarea'

describe('TextArea', () => {
  it('TextArea should support onKeyDown and onPressEnter', () => {
    const fakeHandleKeyDown = jest.fn()
    const fakeHandlePressEnter = jest.fn()

    const wrapper = mount(<TextArea onKeyDown={fakeHandleKeyDown} onPressEnter={fakeHandlePressEnter} />)
    wrapper.find('textarea').simulate('keydown', {keyCode: 65})
    expect(fakeHandleKeyDown).toHaveBeenCalledTimes(1)
    expect(fakeHandlePressEnter).toHaveBeenCalledTimes(0)
    wrapper.find('textarea').simulate('keydown', {keyCode: 13})
    expect(fakeHandleKeyDown).toHaveBeenCalledTimes(2)
    expect(fakeHandlePressEnter).toHaveBeenCalledTimes(1)
  })

  it('TextArea should support maxLength', () => {
    const wrapper = mount(<TextArea maxLength={10} />)
    expect(wrapper.render()).toMatchSnapshot()
  })

  it('TextArea should support disable', () => {
    const wrapper = mount(<TextArea disabled />)
    expect(wrapper.render()).toMatchSnapshot()
  })
})
