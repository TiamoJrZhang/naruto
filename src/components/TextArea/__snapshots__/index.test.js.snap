// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea TextArea should support disable 1`] = `
<div>
  <textarea
    disabled=""
  />
</div>
`;

exports[`TextArea TextArea should support maxLength 1`] = `
<div>
  <textarea
    maxlength="10"
  />
</div>
`;
